scipy/stats/_distn_infrastructure.py

```diff
@@ -3237,6 +3237,9 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
             raise ValueError("xk and pk need to have the same length.")
         if not np.allclose(np.sum(pk), 1):
             raise ValueError("The sum of provided pk is not 1.")
+        if np.any(np.asarray(xk).astype(int) != xk):
+            raise ValueError("xk must be integers. If you really want floats, "
+                             "use xk as integer indices into the float array.")
```

Comment:
I meant to mean 

```
In [11]: xk = np.array([1.5, 2.5, 8.9])

In [12]: pk = np.array([0.3, 0.3, 0.4])

In [13]: rv = stats.rv_discrete(values=(list(range(xk.size)), pk))

In [14]: x = rv.rvs(size=11)

In [15]: xk[x]
Out[15]: array([ 1.5,  2.5,  2.5,  8.9,  1.5,  1.5,  2.5,  8.9,  2.5,  1.5,  2.5])
```

Source: http://stackoverflow.com/questions/37467486/getting-scipys-rv-discrete-to-work-with-floating-point-values